### DISCUSSION TITLE
How to overcome VPN rules for SSL connection.

### DISCUSSION THREAD

**Discussion by PStajemniczadal** [ranaroussi/yfinance#2081] [Q&A]
I have problem with getting any data as my provider uses ZSACLER VPN, I am using yfinance.__version__: 0.2.44

```
import yfinance
ticker = yfinance.Ticker("AAPL")
```
returns
Failed to get ticker 'AAPL' reason: HTTPSConnectionPool(host='fc.yahoo.com', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))

I get similar error when i try to use requests library 
```
import requests
r = requests.get("https://fc.yahoo.com")
```
returns
requests.exceptions.SSLError: HTTPSConnectionPool(host='fc.yahoo.com', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))
```r = requests.get("https://fc.yahoo.com:8443")```
requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='fc.yahoo.com', port=8443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000023CDB5D4490>, 'Connection to fc.yahoo.com timed out. (connect timeout=None)'))

I can add, that when I call https://fc.yahoo.com in webrowser its redirected to opening https://www.yahoo.com/. I can open https://finance.yahoo.com/ in browser. 

I have no access to ZSCALER definition. 
Any help appreciated. I look  for tips how to overcome this problem.

**Comment by jpkern0** [comment]
I have the same problem with ExpressVPN. No solution yet but yfinance worked when I turned off the VPN and ran yf.Ticker. This continued to work even after turning the VPN back on. I'm going to try split tunneling to circumvent the VPN in the short term, but that's not a good long term solution.
